source/fuzz/transformation_move_block_down.h

```diff
@@ -0,0 +1,48 @@
+#include <utility>
+
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_MOVE_BLOCK_DOWN_H_
+#define SOURCE_FUZZ_TRANSFORMATION_MOVE_BLOCK_DOWN_H_
+
+#include "source/fuzz/fact_manager.h"
+#include "source/fuzz/protobufs/spirvfuzz_protobufs.h"
+#include "source/opt/ir_context.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace transformation {
+
+// - |block_id| must be the id of a block b in the given module.
+// - b must not be the first nor last block appearing, in program order,
+//   in a function.
+// - b must not dominate the block that follows it in program order.
+bool IsApplicable(const protobufs::TransformationMoveBlockDown& message,
```

Comment:
Isn't this going to conflict with the transform_split_block code? I think you need to add another namespace, or wrap these things in classes. Otherwise you've got two spvtools::fuzz::transform::IsApplicable and Apply methods.

Reply:
It doesn't conflict because the types of their first arguments are different.  In the case of transform_split_block the first argument is protobufs::TransformationSplitBlock; in this case it is protobufs::TransformMoveBlockDown, and other transformations will IsApplicable and Apply methods but with their own protobuf message type as the first argument.  I was planning to use a single namespace, "transformation", for these methods.

What we really want is a Transformation interface with Apply and IsApplicable methods, and to then have an implementation of this interface for each kind of transformation.  Protobufs doesn't generate code that way, though, hence instead overloading on the first argument.